src/peergos/server/storage/GarbageCollector.java

```diff
@@ -0,0 +1,146 @@
+package peergos.server.storage;
+
+import peergos.server.util.*;
+import peergos.shared.cbor.*;
+import peergos.shared.crypto.hash.*;
+import peergos.shared.io.ipfs.multihash.*;
+import peergos.shared.storage.*;
+
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.*;
+import java.util.logging.*;
+
+public class GarbageCollector implements ContentAddressedStorage {
+
+    private static final long MAX_WAIT_FOR_TRANSACTION_MILLIS = 10_000;
+
+    private final ContentAddressedStorage target;
+    private final long gcPeriodMillis;
+    private final Object gcLock = new Object();
```

Comment:
Can you add a comment about what state this lock is guarding?

Reply:
done